### ISSUE TITLE
✨ API - Talent Event and Development Program

### ISSUE THREAD

**Issue by petertgiles** [GCTC-NTGC/gc-digital-talent#12301]
## ✨ Feature

Build the API for the "talent event" and "development program" models. 

## 🕵️ Details

We need two new models:

1. TalentEvent - represents a community opportunity that employees can be nominated to be nominees for
2. DevelopmentProgram - represents a program run by a community for talent development

We need several mutations:
1. CreateTalentEvent - makes a new model
2. UpdateTalentEvent 
   - updates user-changeable fields and relationships (not timestamps)
   - no state limitations: user-changeable fields can be changed at any time
4. CreateDevelopmentProgram - makes a new model
5. UpdateDevelopmentProgram - updates existing model

Events can be archived so they are no longer visible to public, but visible to community admins in admin pages

## 🎨 Design

@JoshBeveridge 

https://www.figma.com/design/k9lJ77FgLFvpO9L9tVyzID/Talent-nomination-(Employees)?node-id=314-60920&node-type=frame&t=RdnKkm2vLT8XdA2B-0

## 📸 Screenshot

![image](https://github.com/user-attachments/assets/6b1fd49d-5d0e-43e1-bcce-cb343a2c6e9b)

## 🙋‍♀️ Proposed Implementation

```graphql
type TalentEvent
{
  name: LocalizedString
  description: LocalizedString
  open_date: Date
  close_date: Date
  community: Community! @belongsTo
  learn_more_url: LocalizedString
  classifications: [Classifications!] @belongsToMany
  image_url: String
  include_leadership_competencies: Boolean
  close_date_changed_at: Date
  developmentPrograms: [DevelopmentProgram!] @belongsToMany
  calculated_status: TalentEventStatus (Open, Closed)
}

type DevelopmentProgram
{
  name: LocalizedString
  description: LocalizedString
  community: Community! @belongsTo
  classifications: [Classifications!] @belongsToMany
}
```

## ✅ Acceptance Criteria

- [ ] Only community admin (for the community) can create a new talent event
- [ ] Only community admin can update any talent event
- [ ] When closed_at is changed on an open pool, "deadline_changed" becomes true, notification fired to all submitters with draft nominations in event (separate issue)
- [ ] Any user can query for a single talent event by ID
- [ ] Any user can query for all talent events
- [ ] All model changes (including relationships) tracked in the logging table

## 🛑 Blockers

```[tasklist]
### Blocked By
- 
```
